source/geod24/LocalRest.d

```diff
@@ -2503,3 +2519,69 @@ unittest
     assert(extnode.required() == 42);
     assertThrown!ClientException(extnode.optional());
 }
+
+/// Graceful handling of shut down nodes
+unittest
+{
+    static import geod24.concurrency;
+    import core.atomic : atomicLoad, atomicStore;
+
+    static interface API
+    {
+        void test () @safe;
+    }
+
+    static shared bool connection_closed;
+    __gshared Listener!API api_1;
+    __gshared Listener!API api_2;
+
+    static class Node1 : API
+    {
+        Timer timer;
+        RemoteAPI!API api;
+
+        @trusted:
+        public override void test ()
+        {
+            this.api = new RemoteAPI!API(api_2);
+            this.timer = setTimer(10.msecs, &infinite, true);
+        }
+
+        public void infinite () @trusted
+        {
+            try
+            {
+                this.api.test();
+            }  // can handle it explicitly, or let the base class handle it
+            catch (LocalRestConnectionClosed ex)
+            {
+                atomicStore(connection_closed, true);
+                throw ex;  // base class will kill the timer
+            }
+        }
+    }
+
+    static class Node2 : API
+    {
+        @safe:
+        public override void test ()
+        {
+        }
+    }
+
+    scope (exit) thread_joinAll();
+
+    auto node_1 = RemoteAPI!API.spawn!Node1();
+    auto node_2 = RemoteAPI!API.spawn!Node2();
+    api_1 = node_1.ctrl.listener();
+    api_2 = node_2.ctrl.listener();
+
+    node_1.test();
+    node_2.ctrl.shutdown();
+    Thread.sleep(1.seconds);  // trigger calling shut down node
+    node_1.ctrl.shutdown();
+
+    size_t count; // up to 3 seconds wait
+    while (!atomicLoad(connection_closed))
+        assert(count < 300); count++; Thread.sleep(10.msecs);
```

Comment:
So this doesnt need curly braces? D is a weird language.

Reply:
You know what happened? The stupid coverage bot failed my PR because 3 lines weren't covered. So I put them all on the same line. And then I removed the braces. I need sleep.